lib/connection.ts

```diff
@@ -600,12 +600,12 @@ export class Connection extends Entity {
    * @returns {void} void
    */
   private _initializeEventListeners(): void {
-    for (const eventName in ConnectionEvents) {
-      this._connection.on(ConnectionEvents[eventName], (context) => {
+    for (const eventName in Object.values(ConnectionEvents)) {
```

Comment:
So it looks like Object.values is an ES2017 feature, which is why it doesn't work in node.js 6.

You could do something like this instead, but it is more verbose.

```suggestion
    for (const eventName of Object.keys(ConnectionEvents) as Array<keyof typeof ConnectionEvents>) {
```

Reply:
> So it looks like Object.values is an ES2017 feature, which is why it doesn't work in node.js 6.

But, thats the funny thing. Node 6 is fine. Its Node 8 and 10 that has a problem with Object.Values

Reply:
Well, npm test isn't actually running any code, so while Node 6 might compile fine, I don't think it will work at runtime.

That said I don't know why the compiler doesn't complain with node 6 and not 8/10. I'd expect it to complain for all of them since the target is ES2015 and not ES2017.